banyand/protector/protector.go

```diff
@@ -37,6 +37,21 @@ import (
 
 var scope = observability.RootScope.SubScope("memory_protector")
 
+// MemoryProtector is an interface for monitoring and limiting memory usage to prevent OOM.
+type MemoryProtector interface {
```

Comment:
```suggestion
// Memory is an interface for monitoring and limiting memory usage to prevent OOM.
type Memory interface {
```

Reply:
and convert the struct's name from `Memory` to `memory`

Reply:
Yes, the role of the interface is to conceal the structure. If you introduce the interface, the exported structure no longer makes sense.